test/oas2json.test.js

```diff
@@ -127,5 +128,77 @@ tap.test('oas2json', async t => {
       )
       t.end()
     })
+
+    t.test(
+      'should convert components.schemas automatically if $ref exists in targeted property',
+      t => {
+        runCommand(inputPath, outputPath, 'components.requestBodies')
+
+        const generatedDirs = fs.readdirSync(outputPath)
+        t.match(
+          generatedDirs,
+          ['components.requestBodies', 'components.schemas'],
+          'generates the expected directories'
+        )
+
+        const petReqBody = fs.readJSONSync(
+          resolveFromPackageRoot(
+            outputPath,
+            'components.requestBodies/Pet.json'
+          )
+        )
+        t.same(
+          petReqBody,
+          {
+            description: 'Pet object that needs to be added to the store',
+            content: {
+              'application/json': {
+                schema: {
+                  $ref: '../components.schemas/Pet.json'
+                }
+              },
+              'application/xml': {
+                schema: {
+                  $ref: '../components.schemas/Pet.json'
+                }
+              }
+            },
+            title: 'Pet',
+            $id: 'Pet.json'
+          },
+          'generates the expected JSON with $ref set correctly'
+        )
+
+        t.end()
+      }
+    )
+
+    t.test(
+      'should dynamically find the name when converting an array property',
+      t => {
```

Comment:
use the async pattern to write tests, using t.end is a leftover from the past besides being error prone. consider changing other tests in this file to reflect this, but thanks for keeping the consistency

Reply:
Sure, no problem - I'll get that updated